packages/pyright-internal/src/analyzer/typePrinter.ts

```diff
@@ -245,11 +253,14 @@ export function printType(
                 // If it's a Callable with a ParamSpec, use the
                 // Callable notation.
                 const parts = printFunctionParts(type, printTypeFlags, returnTypeCallback, recursionTypes);
-                const paramSignature = `(${parts[0].join(', ')})`;
+                const paramSignature = annotationFormat ? `${parts[0].join(', ')}` : `(${parts[0].join(', ')})`; // no brackets if it's an annotation
+
                 if (FunctionType.isParamSpecValue(type)) {
                     return paramSignature;
                 }
-                const fullSignature = `${paramSignature} -> ${parts[1]}`;
+                const fullSignature = annotationFormat
+                    ? `Callable[${paramSignature}, ${parts[1]}]`
```

Comment:
There are some signatures that cannot be represented by `Callable`. For example, signatures that include position-only separators, keyword-only arguments, *args, **kwargs, ParamSpecs, TypeVarTuples, and parameters with default arguments. It's not clear we should do in those cases. Perhaps fall back to `Callable[..., <return>]`?